be/src/service/service_be/lake_service.cpp

```diff
@@ -182,12 +182,14 @@ void LakeServiceImpl::publish_version(::google::protobuf::RpcController* control
                 auto score = compaction_score(metadata);
                 std::lock_guard l(response_mtx);
                 response->mutable_compaction_scores()->insert({tablet_id, score});
+                Status::OK().to_protobuf(response->mutable_status());
```

Comment:
Initialize the response status to ok before entering the for loop. This prevents later success from overwriting and dropping the error status, since currently any error will be overridden by a subsequent success assignment.